src/v/config/property.h

```diff
@@ -900,13 +900,7 @@ class enum_property : public property<T> {
 private:
     ss::sstring help_text() const {
         // String-ize the available values
-        std::vector<std::string> str_values;
-        str_values.reserve(_values.size());
-        std::transform(
-          _values.begin(),
-          _values.end(),
-          std::back_inserter(str_values),
-          [](const T& v) { return fmt::format("{}", v); });
+        std::vector<ss::sstring> str_values = enum_values();
```

Comment:
We don't need the intermediate container, right?
```suggestion
        return fmt::format("Must be one of {}", fmt::join(_values, ","));
```
Oh, I see, in the next commit, we rely on the return value of enum_values().

Still, perhaps:
```suggestion
        return fmt::format("Must be one of {}", fmt::join(enum_values(), ","));
```